.github/workflows/pr_testing_check.yaml

```
@@ -0,0 +1,119 @@
+name: Run testcases that changed on this PR
+
+on: pull_request
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4.1.2
+
+    - name: Set up Python
+      id: setup-python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.12'
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pylint
```

Do we know if there is a way to cache this, so we dont need to execute it everytime we run the action,

```suggestion
  - name: Cache Python dependencies
    uses: actions/cache@v2
    with:
      path: ~/.cache/pip
      key: ${{ runner.os }}-pip
      restore-keys: |
        ${{ runner.os }}-pip

    - name: Install dependencies
      run: |
        python -m pip install --upgrade pip
        pip install pylint
```

---

@CubeVic I think we could use the Cache Action (https://github.com/actions/cache) 

Here is an example of how it could be done (I haven't tested it myself)

- name: Cache pip
      uses: actions/cache@v3
      with:
        path: ~/.cache/pip
        key: ${{ runner.os }}-${{ hashFiles('**/requirements.txt') }}